main.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_map(&proc)
+    temp = []
+    (0..length - 1).each do |i|
+      temp << [proc.call(self[i])] if proc
+      temp << yield(self[i]) if block_given? && !proc
+    end
+    temp
+  end
+
+  def my_each_with_index
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    temp = []
+    (0..length - 1).each do |i|
+      temp << self[i] if yield(self[i])
+    end
+    temp
+  end
+
+  def my_any?
+    each do |item|
+      return true if block_given? && yield(item)
+      return true if item && !block_given?
```

Comment:
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern